### ISSUE TITLE
RuntimeError: Could not load the model status=4 message=Allocation Failure

### ISSUE THREAD

**Issue by Tagussan** [aws/aws-neuron-sdk#369]
Hi,

I have successfully compiled a model with `torch.neuron.trace`. When tracing, I got many warning message like `12/03/2021 07:46:45 AM WARNING [Stargazer]: SBUF DMA write size != 0 mod 4: SBUF address=0x8482192, size=54`

After I got the neuron model file, I loaded the compiled model with `torch.jit.load` and run inference. However, I got the following error message:
```
2021-Dec-03 08:02:36.0093  5110:5110  ERROR  TDRV:dma_ring_allocate                       No more pages for ring type: 1
2021-Dec-03 08:02:36.0093  5110:5110  ERROR  TDRV:kbl_model_add                           create_data_refill_rings() error
2021-Dec-03 08:02:36.0120  5110:5110  ERROR  NMGR:dlr_kelf_stage                          Failed to load subgraph
2021-Dec-03 08:02:36.0120  5110:5110  ERROR  NMGR:stage_kelf_models                       Failed to stage graph: kelf-a.json to NeuronCore
2021-Dec-03 08:02:36.0125  5110:5110  ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: 1.7.3.0+c02f9fde4-/tmp/tmpt83tcdeo, err: 4
Traceback (most recent call last):
  File "test.py", line 35, in <module>
    print(model_neuron_loaded(data_in))                                                                                                                                                                                                               File "/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/__torch__/torch_neuron/convert.py", line 8, in forward
    argument_1: Tensor) -> Tensor:
    _0 = getattr(self, "_NeuronGraph#0")
    _1 = ops.neuron.forward_1([argument_1], CONSTANTS.c0, CONSTANTS.c1, CONSTANTS.c2)
         ~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return _1

Traceback of TorchScript, original code (most recent call last):
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/decorators.py(412): forward
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py(534): _slow_forward
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py(548): __call__
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/graph.py(546): __call__
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/graph.py(205): run_op
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/graph.py(194): __call__
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/convert.py(220): forward
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py(534): _slow_forward
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py(548): __call__
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/jit/__init__.py(1027): trace_module
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/jit/__init__.py(875): trace
/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/convert.py(186): trace
test.py(31): <module>
RuntimeError: Could not load the model status=4 message=Allocation Failure
```

I'm using torch-neuron `1.5.1.2.0.392.0` and neuron-cc `1.7.3.0+c02f9fde4`.
How can I fix this error? Thanks.

**Comment by Tagussan** [comment]
Here's the code to reproduce: https://gist.github.com/Tagussan/29027d77925c0a3f3cbceb36d4c011e1

**Comment by mrnikwaws** [comment]
Hi @Tagussan,

Thanks for reporting this error.  The model you are using appears to have exposed a bug in Neuron compilation, which results in a runtime failure.  We are investigating a fix and will update you once we have more information.

**Comment by Tagussan** [comment]
Hi @mrnikwaws,
I appreciate your quick responce and looking forward to seeing updates. Thank you!

**Comment by aws-taylor** [comment]
Hello @Tagussan,

  There have been some fixes in the most recent version of our runtime driver that we believe may fix this issue. Can you check which version of `aws-neuron-dkms` you are using, and upgrade to the most recent version of possible?

-Taylor

**Comment by Tagussan** [comment]
Hi @aws-taylor 

I'm using `aws-neuron-dkms` 2.2.6.0 now. I followed the update instruction described [here](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-intro/pytorch-setup/pytorch-update.html) and there was no update applied.

**Comment by Tagussan** [comment]
FYI: Since neuron-sdk v1.16.2 is released, I updated the SDK and run the compilation again. `neuron-cc -V` is `1.8.2.0+5b63f6edb`.  The version of `torch-neuron` is `1.5.1.2.0.468.0`.
Although the compilation successfully finishes, I got the same runtime error again.

**Comment by shebbur-aws** [comment]
Hi @Tagussan,

The fix for this was not included in any official release yet. Please do expect a fix in the upcoming release.

Thanks,
Shruthi.

**Comment by aws-joshim** [comment]
@Tagussan we will update this ticket with the release details that includes the fix once the release is out

**Comment by Tagussan** [comment]
Hi @aws-joshim (and Inferentia team),
Happy new year and thank you very much for your support during the past year. I look forward to continuing work with you in 2022. I will wait for the next release.

**Comment by awsrjh** [comment]
Hi

This has been resolved in the recent 1.17.0 [release](https://github.com/aws/aws-neuron-sdk/releases/tag/v1.17.0). 

Please find more information here: 

https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/index.html#neuron-1-17-0-01-20-2022
